### PR TITLE
Dockerfile that builds cuda kernels

### PR DESCRIPTION
**Pull request by NickLucche**
Hey, first of all thanks for sharing your work, it looks promising!
I had a bit of struggle to get it to run with a more recent stack due to (mainly) the absence of the `THC.h` header in modern pytorch releases, needed to build the cuda functions.
I believe providing a very simple Dockerfile such as this could be helpful to get people started right away, as it includes the right cuda tools (devel image, so it comes with things like nvcc out of the box).

The image takes care of building the kernels on the platform, as long as `nvidia-container-runtime` is the default runtime:

```bash
# /etc/docker/daemon.json
{
    "runtimes": {
        "nvidia": {
            "path": "/usr/bin/nvidia-container-runtime",
            "runtimeArgs": []
         } 
    },
    "default-runtime": "nvidia" 
}
```

### CODE REVIEW

**Comment by youmi-zym** [comment]
Hi,

Thanks for the great job! Did you ever evaluate on NYUv2 test dataset to get the metric results? It'd be super cool if you could provide that, just to double-check! Thanks again!

**Comment by NickLucche** [comment]
Sure I can run the evaluation within the container

**Comment by NickLucche** [comment]
Here's my result running evaluation on NYUv2 with:
`python main.py --dir_data /data/nyudepthv2/ --data_name NYU  --split_json ../data_json/nyu.json  --gpus 0 --max_depth 10.0 --num_sample 500   --test_only --pretrain /data/nyuv2.pt --save test_restults`

```
Metric   |  RMSE: 0.09013  MAE: 0.03519  iRMSE: 0.01378  iMAE: 0.00508  REL: 0.01189  D^1: 0.99585  D^2: 0.99940  D^3: 0.99988  D102: 0.87466  D105: 0.95325  D110: 0.98049
```

I can't quite find a reference for all these figures, maybe you could quickly double check (looks good to me tbh, solid numbers).

**Comment by youmi-zym** [comment]
Awesome, thanks! Let me merge it.
